CDCgov/prime-reportstream
# Added support to convert timestamps to date times in the UP

This PR adds support convert timestamps to date times in the UP for HL7 v2 messages using the `convertTimestampToDateTime` receiver translation setting.

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. Check out and build the branch
2. Add this setting `convertTimestampToDateTime: "MSH-7,SPM-18"` to the `DEV_FULL_ELR` receiver translation settings.
3. run `./gradlew reloadSettings`
4. Send a test report using the attached Postman collection
[Simple Report Bug Fix.postman_collection.txt](https://github.com/CDCgov/prime-reportstream/files/11870198/Simple.Report.Bug.Fix.postman_collection.txt)
5. Verify the request was accepted
6. Open the generated HL7 file in the local SFTP folder
7. Verify that MSH-7 and SPM-18 dates are using this date format: `yyyyMMddHHmmss`

## Changes
- Added support to convert timestamps to date time for HL7 v2 messages

## Checklist

### Testing


## Linked Issues
- #11131
